components/batt/src/batt.c

```diff
@@ -0,0 +1,40 @@
+#include "batt.h"
+
+#include "firmware-base/state-machine.h"
+#include <ember_taskglue.h>
+#include <freertos/FreeRTOS.h>
+#include <freertos/task.h>
+#include <opencan_rx.h>
+#include <opencan_tx.h>
+
+#include <math.h>
+#include <string.h>
+
+// hypothetical battery percent data
+typedef struct {
```

Comment:
We don't need to make a CAN data struct, this is already generated by OpenCAN, we just have to implement a populate function.